Url template editor

streamich opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/69413
Closes https://github.com/elastic/kibana/issues/83328

This PR replaces URL Drilldown textarea for URL template editing with Monaco editor. This enables:

- Syntax highlighting for Handlebars variables.
- Syntax highlighting for different URL parts.
- Autocompletion for Handlebars variables.
- Description text for each variable inside autocomplete popup.
- Supports dark mode.
- Previews variable values in autocomplete popup.
- Sample URL instead of placeholder.

Simple URL:

![image](https://user-images.githubusercontent.com/9773803/106901682-5b066380-66f8-11eb-9c8a-629f31c7bdd7.png)

With query params:

![image](https://user-images.githubusercontent.com/9773803/106901800-6fe2f700-66f8-11eb-8254-33f42538ed17.png)

Complex long link:

![image](https://user-images.githubusercontent.com/9773803/106901851-7d987c80-66f8-11eb-9a28-bbd903f62792.png)

With *if-statement*:

![image](https://user-images.githubusercontent.com/9773803/106901925-8d17c580-66f8-11eb-931b-34a0f80700c3.png)

Autocompletion:

![image](https://user-images.githubusercontent.com/9773803/106902002-9e60d200-66f8-11eb-86ca-8499f8554901.png)

Description about each variable:

![image](https://user-images.githubusercontent.com/9773803/106902070-b173a200-66f8-11eb-81e8-de7bd68c29fd.png)



### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [x] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/)
* Commit: 6ccee8e9abf49600f59140cf2aa3a05a462ed784
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup11](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/execution/node/684/log)
* [Execute xpack-kibana-ciGroup5](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/execution/node/634/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/","number":"99359","commit":"6ccee8e9abf49600f59140cf2aa3a05a462ed784"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/)
* Commit: 856603925932955d1de033551d52de318ceba8d8
* This comment will update when the build is complete

### Failed CI Steps
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/execution/node/465/log)
* [Execute oss-pluginFunctional](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/execution/node/642/log)

### History
* :broken_heart: [Build #99359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/) failed 6ccee8e9abf49600f59140cf2aa3a05a462ed784

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/","number":"99359","commit":"6ccee8e9abf49600f59140cf2aa3a05a462ed784"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/","number":"100277","commit":"856603925932955d1de033551d52de318ceba8d8"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/)
* Commit: 3d53855f70e6cd1e26541c7959ab07fb9d20c943
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/execution/node/106/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/execution/node/112/log)

### History
* :broken_heart: [Build #100277](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/) failed 856603925932955d1de033551d52de318ceba8d8
* :broken_heart: [Build #99359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/) failed 6ccee8e9abf49600f59140cf2aa3a05a462ed784

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/","number":"99359","commit":"6ccee8e9abf49600f59140cf2aa3a05a462ed784"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/","number":"100277","commit":"856603925932955d1de033551d52de318ceba8d8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/","number":"101739","commit":"3d53855f70e6cd1e26541c7959ab07fb9d20c943"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/)
* Commit: 0cdf2c4423ba0bcf163cf5c1373a758a01af23ef
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/execution/node/106/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/execution/node/112/log)

### History
* :broken_heart: [Build #101739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/) failed 3d53855f70e6cd1e26541c7959ab07fb9d20c943
* :broken_heart: [Build #100277](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/) failed 856603925932955d1de033551d52de318ceba8d8
* :broken_heart: [Build #99359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/) failed 6ccee8e9abf49600f59140cf2aa3a05a462ed784

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/","number":"99359","commit":"6ccee8e9abf49600f59140cf2aa3a05a462ed784"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/","number":"100277","commit":"856603925932955d1de033551d52de318ceba8d8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/","number":"101739","commit":"3d53855f70e6cd1e26541c7959ab07fb9d20c943"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/","number":"101897","commit":"0cdf2c4423ba0bcf163cf5c1373a758a01af23ef"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/)
* Commit: d10ba44bd754829232eac480cf33f90ace88fbd1
* This comment will update when the build is complete

### Failed CI Steps
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/execution/node/106/log)
* [Setup Build Environment and Dependencies](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/execution/node/112/log)

### History
* :broken_heart: [Build #101897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/) failed 0cdf2c4423ba0bcf163cf5c1373a758a01af23ef
* :broken_heart: [Build #101739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/) failed 3d53855f70e6cd1e26541c7959ab07fb9d20c943
* :broken_heart: [Build #100277](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/) failed 856603925932955d1de033551d52de318ceba8d8
* :broken_heart: [Build #99359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/) failed 6ccee8e9abf49600f59140cf2aa3a05a462ed784

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/","number":"99359","commit":"6ccee8e9abf49600f59140cf2aa3a05a462ed784"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/","number":"100277","commit":"856603925932955d1de033551d52de318ceba8d8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/","number":"101739","commit":"3d53855f70e6cd1e26541c7959ab07fb9d20c943"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/","number":"101897","commit":"0cdf2c4423ba0bcf163cf5c1373a758a01af23ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/","number":"101945","commit":"d10ba44bd754829232eac480cf33f90ace88fbd1"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/)
* Commit: 1f8d552fa298fbd2f3ccca3009816068f654ed19
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/execution/node/503/log)

### History
* :broken_heart: [Build #101945](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/) failed d10ba44bd754829232eac480cf33f90ace88fbd1
* :broken_heart: [Build #101897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/) failed 0cdf2c4423ba0bcf163cf5c1373a758a01af23ef
* :broken_heart: [Build #101739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/) failed 3d53855f70e6cd1e26541c7959ab07fb9d20c943
* :broken_heart: [Build #100277](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/) failed 856603925932955d1de033551d52de318ceba8d8
* :broken_heart: [Build #99359](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/) failed 6ccee8e9abf49600f59140cf2aa3a05a462ed784

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99359/","number":"99359","commit":"6ccee8e9abf49600f59140cf2aa3a05a462ed784"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/","number":"100277","commit":"856603925932955d1de033551d52de318ceba8d8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/","number":"101739","commit":"3d53855f70e6cd1e26541c7959ab07fb9d20c943"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/","number":"101897","commit":"0cdf2c4423ba0bcf163cf5c1373a758a01af23ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/","number":"101945","commit":"d10ba44bd754829232eac480cf33f90ace88fbd1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/","number":"102677","commit":"1f8d552fa298fbd2f3ccca3009816068f654ed19"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/)
* Commit: fa23a5923a1c768543749ab3ee8363e2639007e6
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/execution/node/505/log)

### History
* :broken_heart: [Build #102677](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/) failed 1f8d552fa298fbd2f3ccca3009816068f654ed19
* :broken_heart: [Build #101945](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/) failed d10ba44bd754829232eac480cf33f90ace88fbd1
* :broken_heart: [Build #101897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/) failed 0cdf2c4423ba0bcf163cf5c1373a758a01af23ef
* :broken_heart: [Build #101739](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/) failed 3d53855f70e6cd1e26541c7959ab07fb9d20c943
* :broken_heart: [Build #100277](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/) failed 856603925932955d1de033551d52de318ceba8d8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100277/","number":"100277","commit":"856603925932955d1de033551d52de318ceba8d8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101739/","number":"101739","commit":"3d53855f70e6cd1e26541c7959ab07fb9d20c943"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/","number":"101897","commit":"0cdf2c4423ba0bcf163cf5c1373a758a01af23ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/","number":"101945","commit":"d10ba44bd754829232eac480cf33f90ace88fbd1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/","number":"102677","commit":"1f8d552fa298fbd2f3ccca3009816068f654ed19"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/","number":"103411","commit":"fa23a5923a1c768543749ab3ee8363e2639007e6"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/)
* Commit: 2e30c236d72cad055746a224376250c0a4004cf0
* This comment will update when the build is complete

### Failed CI Steps
* [Lint: eslint](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/execution/node/467/log)

### History
* :green_heart: [Build #103422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/) succeeded 6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb
* :broken_heart: [Build #103411](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/) failed fa23a5923a1c768543749ab3ee8363e2639007e6
* :broken_heart: [Build #102677](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/) failed 1f8d552fa298fbd2f3ccca3009816068f654ed19
* :broken_heart: [Build #101945](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/) failed d10ba44bd754829232eac480cf33f90ace88fbd1
* :broken_heart: [Build #101897](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/) failed 0cdf2c4423ba0bcf163cf5c1373a758a01af23ef

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101897/","number":"101897","commit":"0cdf2c4423ba0bcf163cf5c1373a758a01af23ef"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/","number":"101945","commit":"d10ba44bd754829232eac480cf33f90ace88fbd1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/","number":"102677","commit":"1f8d552fa298fbd2f3ccca3009816068f654ed19"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/","number":"103411","commit":"fa23a5923a1c768543749ab3ee8363e2639007e6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/","number":"103422","commit":"6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/","number":"103833","commit":"2e30c236d72cad055746a224376250c0a4004cf0"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/)
* Commit: 26176505885973a7657b4a4701d9f4cda3347924
* [Documentation Changes](https://kibana_88577.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f3a9c763dfa08972b8106fac248456b4b3f20086) | [after](https://github.com/elastic/kibana/commit/26176505885973a7657b4a4701d9f4cda3347924) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 320 | 333 | +13 |
| `uiActionsEnhanced` | 120 | 122 | +2 |
| `urlDrilldown` | 29 | 32 | +3 |
| total |  |  | +18 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f3a9c763dfa08972b8106fac248456b4b3f20086) | [after](https://github.com/elastic/kibana/commit/26176505885973a7657b4a4701d9f4cda3347924) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 348.9KB | 349.0KB | +146.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f3a9c763dfa08972b8106fac248456b4b3f20086) | [after](https://github.com/elastic/kibana/commit/26176505885973a7657b4a4701d9f4cda3347924) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 119.4KB | 126.3KB | +7.0KB |
| `uiActionsEnhanced` | 226.3KB | 226.1KB | -246.0B |
| `urlDrilldown` | 42.1KB | 51.2KB | +9.1KB |
| total |  |  | +15.8KB |

### History
* :broken_heart: [Build #103833](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/) failed 2e30c236d72cad055746a224376250c0a4004cf0
* :green_heart: [Build #103422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/) succeeded 6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb
* :broken_heart: [Build #103411](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/) failed fa23a5923a1c768543749ab3ee8363e2639007e6
* :broken_heart: [Build #102677](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/) failed 1f8d552fa298fbd2f3ccca3009816068f654ed19
* :broken_heart: [Build #101945](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/) failed d10ba44bd754829232eac480cf33f90ace88fbd1

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101945/","number":"101945","commit":"d10ba44bd754829232eac480cf33f90ace88fbd1"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/","number":"102677","commit":"1f8d552fa298fbd2f3ccca3009816068f654ed19"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/","number":"103411","commit":"fa23a5923a1c768543749ab3ee8363e2639007e6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/","number":"103422","commit":"6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/","number":"103833","commit":"2e30c236d72cad055746a224376250c0a4004cf0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/","number":"103870","commit":"26176505885973a7657b4a4701d9f4cda3347924"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/)
* Commit: ab442a6d734094a27becb4b7107e66d30cc8208b
* [Documentation Changes](https://kibana_88577.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/b058f7852b74fd22f396c367dc47cf4a526b4daa) | [after](https://github.com/elastic/kibana/commit/ab442a6d734094a27becb4b7107e66d30cc8208b) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 320 | 333 | +13 |
| `uiActionsEnhanced` | 120 | 122 | +2 |
| `urlDrilldown` | 29 | 32 | +3 |
| total |  |  | +18 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/b058f7852b74fd22f396c367dc47cf4a526b4daa) | [after](https://github.com/elastic/kibana/commit/ab442a6d734094a27becb4b7107e66d30cc8208b) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 348.9KB | 349.0KB | +146.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/b058f7852b74fd22f396c367dc47cf4a526b4daa) | [after](https://github.com/elastic/kibana/commit/ab442a6d734094a27becb4b7107e66d30cc8208b) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 119.4KB | 126.3KB | +7.0KB |
| `uiActionsEnhanced` | 226.3KB | 226.1KB | -246.0B |
| `urlDrilldown` | 42.1KB | 51.2KB | +9.1KB |
| total |  |  | +15.8KB |

### History
* :green_heart: [Build #103870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/) succeeded 26176505885973a7657b4a4701d9f4cda3347924
* :broken_heart: [Build #103833](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/) failed 2e30c236d72cad055746a224376250c0a4004cf0
* :green_heart: [Build #103422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/) succeeded 6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb
* :broken_heart: [Build #103411](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/) failed fa23a5923a1c768543749ab3ee8363e2639007e6
* :broken_heart: [Build #102677](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/) failed 1f8d552fa298fbd2f3ccca3009816068f654ed19

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/102677/","number":"102677","commit":"1f8d552fa298fbd2f3ccca3009816068f654ed19"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/","number":"103411","commit":"fa23a5923a1c768543749ab3ee8363e2639007e6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/","number":"103422","commit":"6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/","number":"103833","commit":"2e30c236d72cad055746a224376250c0a4004cf0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/","number":"103870","commit":"26176505885973a7657b4a4701d9f4cda3347924"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/","number":"104310","commit":"ab442a6d734094a27becb4b7107e66d30cc8208b"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/)
* Commit: 1a525b0adb40528e6ae53135d1c12a605ed2c7f0
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/execution/node/505/log)

### History
* :green_heart: [Build #104310](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/) succeeded ab442a6d734094a27becb4b7107e66d30cc8208b
* :green_heart: [Build #103870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/) succeeded 26176505885973a7657b4a4701d9f4cda3347924
* :broken_heart: [Build #103833](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/) failed 2e30c236d72cad055746a224376250c0a4004cf0
* :green_heart: [Build #103422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/) succeeded 6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb
* :broken_heart: [Build #103411](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/) failed fa23a5923a1c768543749ab3ee8363e2639007e6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103411/","number":"103411","commit":"fa23a5923a1c768543749ab3ee8363e2639007e6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/","number":"103422","commit":"6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/","number":"103833","commit":"2e30c236d72cad055746a224376250c0a4004cf0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/","number":"103870","commit":"26176505885973a7657b4a4701d9f4cda3347924"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/","number":"104310","commit":"ab442a6d734094a27becb4b7107e66d30cc8208b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/","number":"104648","commit":"1a525b0adb40528e6ae53135d1c12a605ed2c7f0"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/)
* Commit: 6f9189438c93e51122acb7050b4a2ddc0b73acc8
* [Documentation Changes](https://kibana_88577.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/2279c06d1e0732ffea9de6f2ef1824eb00613ad5) | [after](https://github.com/elastic/kibana/commit/6f9189438c93e51122acb7050b4a2ddc0b73acc8) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 320 | 333 | +13 |
| `uiActionsEnhanced` | 120 | 122 | +2 |
| `urlDrilldown` | 29 | 33 | +4 |
| total |  |  | +19 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/2279c06d1e0732ffea9de6f2ef1824eb00613ad5) | [after](https://github.com/elastic/kibana/commit/6f9189438c93e51122acb7050b4a2ddc0b73acc8) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 348.9KB | 349.0KB | +146.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/2279c06d1e0732ffea9de6f2ef1824eb00613ad5) | [after](https://github.com/elastic/kibana/commit/6f9189438c93e51122acb7050b4a2ddc0b73acc8) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 119.4KB | 126.8KB | +7.4KB |
| `uiActionsEnhanced` | 226.3KB | 226.1KB | -246.0B |
| `urlDrilldown` | 42.1KB | 51.8KB | +9.7KB |
| total |  |  | +16.9KB |

### History
* :broken_heart: [Build #104648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/) failed 1a525b0adb40528e6ae53135d1c12a605ed2c7f0
* :green_heart: [Build #104310](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/) succeeded ab442a6d734094a27becb4b7107e66d30cc8208b
* :green_heart: [Build #103870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/) succeeded 26176505885973a7657b4a4701d9f4cda3347924
* :broken_heart: [Build #103833](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/) failed 2e30c236d72cad055746a224376250c0a4004cf0
* :green_heart: [Build #103422](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/) succeeded 6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103422/","number":"103422","commit":"6cc699ef4d4fde3199c30c62dc5f19b4e3bceebb"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/","number":"103833","commit":"2e30c236d72cad055746a224376250c0a4004cf0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/","number":"103870","commit":"26176505885973a7657b4a4701d9f4cda3347924"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/","number":"104310","commit":"ab442a6d734094a27becb4b7107e66d30cc8208b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/","number":"104648","commit":"1a525b0adb40528e6ae53135d1c12a605ed2c7f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/","number":"104709","commit":"6f9189438c93e51122acb7050b4a2ddc0b73acc8"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/)
* Commit: 0d3c826108c224a3b84da6b4a6d7ec4f311b231a
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/execution/node/858/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/execution/node/916/log)

### History
* :green_heart: [Build #104709](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/) succeeded 6f9189438c93e51122acb7050b4a2ddc0b73acc8
* :broken_heart: [Build #104648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/) failed 1a525b0adb40528e6ae53135d1c12a605ed2c7f0
* :green_heart: [Build #104310](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/) succeeded ab442a6d734094a27becb4b7107e66d30cc8208b
* :green_heart: [Build #103870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/) succeeded 26176505885973a7657b4a4701d9f4cda3347924
* :broken_heart: [Build #103833](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/) failed 2e30c236d72cad055746a224376250c0a4004cf0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103833/","number":"103833","commit":"2e30c236d72cad055746a224376250c0a4004cf0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/","number":"103870","commit":"26176505885973a7657b4a4701d9f4cda3347924"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/","number":"104310","commit":"ab442a6d734094a27becb4b7107e66d30cc8208b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/","number":"104648","commit":"1a525b0adb40528e6ae53135d1c12a605ed2c7f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/","number":"104709","commit":"6f9189438c93e51122acb7050b4a2ddc0b73acc8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/","number":"104783","commit":"0d3c826108c224a3b84da6b4a6d7ec4f311b231a"}]}
PIPELINE-->

streamich commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServices)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/)
* Commit: c0de84fd76b93d0ad2ed5316bc540dd2759a344a
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/execution/node/858/log)
* [Execute xpack-kibana-ciGroup7](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/execution/node/916/log)

### History
* :broken_heart: [Build #104783](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/) failed 0d3c826108c224a3b84da6b4a6d7ec4f311b231a
* :green_heart: [Build #104709](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/) succeeded 6f9189438c93e51122acb7050b4a2ddc0b73acc8
* :broken_heart: [Build #104648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/) failed 1a525b0adb40528e6ae53135d1c12a605ed2c7f0
* :green_heart: [Build #104310](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/) succeeded ab442a6d734094a27becb4b7107e66d30cc8208b
* :green_heart: [Build #103870](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/) succeeded 26176505885973a7657b4a4701d9f4cda3347924

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/103870/","number":"103870","commit":"26176505885973a7657b4a4701d9f4cda3347924"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/","number":"104310","commit":"ab442a6d734094a27becb4b7107e66d30cc8208b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/","number":"104648","commit":"1a525b0adb40528e6ae53135d1c12a605ed2c7f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/","number":"104709","commit":"6f9189438c93e51122acb7050b4a2ddc0b73acc8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/","number":"104783","commit":"0d3c826108c224a3b84da6b4a6d7ec4f311b231a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/","number":"105042","commit":"c0de84fd76b93d0ad2ed5316bc540dd2759a344a"}]}
PIPELINE-->

mdefazio commented on src/plugins/kibana_react/public/url_template_editor/styles.scss:
```diff
@@ -0,0 +1,5 @@
+.urlTemplateEditor__container {
+  .monaco-editor .lines-content.monaco-editor-background {
+    margin: 8px;
```
Can we use the eui variable `$euiSizeS`?

streamich replied:
Done.

mdefazio commented on x-pack/plugins/drilldowns/url_drilldown/public/lib/url_drilldown.tsx:
```diff
@@ -90,21 +96,30 @@ export class UrlDrilldown implements Drilldown<Config, ActionContext, ActionFact
   public readonly createConfig = () => ({
-    url: { template: '' },
+    url: {
+      template: 'https://example.com/?{{event.key}}={{event.value}}',
```
Thinking more on this, it might be wise to add in 'e.g. https://...' or 'Example: https://...' just to make it more clear that this is essentially placeholder.

streamich replied:
I've added "Example: "

mdefazio reviewed (changes requested):
This looks awesome. Just some super tiny nits regarding sass variables and copy.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/)
* Commit: 5b4fefebb980ee84157def88a3f6c6fa866057db
* This comment will update when the build is complete

### Failed CI Steps
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/execution/node/799/log)
* [Execute xpack-firefox](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/execution/node/889/log)

### History
* :broken_heart: [Build #105042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/) failed c0de84fd76b93d0ad2ed5316bc540dd2759a344a
* :broken_heart: [Build #104783](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/) failed 0d3c826108c224a3b84da6b4a6d7ec4f311b231a
* :green_heart: [Build #104709](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/) succeeded 6f9189438c93e51122acb7050b4a2ddc0b73acc8
* :broken_heart: [Build #104648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/) failed 1a525b0adb40528e6ae53135d1c12a605ed2c7f0
* :green_heart: [Build #104310](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/) succeeded ab442a6d734094a27becb4b7107e66d30cc8208b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104310/","number":"104310","commit":"ab442a6d734094a27becb4b7107e66d30cc8208b"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/","number":"104648","commit":"1a525b0adb40528e6ae53135d1c12a605ed2c7f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/","number":"104709","commit":"6f9189438c93e51122acb7050b4a2ddc0b73acc8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/","number":"104783","commit":"0d3c826108c224a3b84da6b4a6d7ec4f311b231a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/","number":"105042","commit":"c0de84fd76b93d0ad2ed5316bc540dd2759a344a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/","number":"105182","commit":"5b4fefebb980ee84157def88a3f6c6fa866057db"}]}
PIPELINE-->

jbudz reviewed (approved):
operations - storybook aliases LGTM

ppisljar reviewed (changes requested):
autocompletion is a bit annoying, is it possible to tweak this ? 
- autocompletion auto pops up while writing your URL at any point, in between url but also when inside {{}} code brackets writing my if statement. if selected there it will insert nested {{}} which is probably not what we want. I would expect autocompletion to either not pop up when i am inside {{}} or to correctly insert variable name in that case (without nested {{}})
- there is no autocompletion on handlebars operators, would that be possible to add, so that when I am inside {{}} i could see what's possible ?
- accessing documentation is only possible with keyboard+mouse combo. mouse over the dropdown entry doenst show the short description and info icon (this is only possible with keyboard). there seems to be no way to select info icon (expand info) with keyboard only (clicking the info icon is only possible with mouse). also there seems to be no way to close the info popup with keyboard (only possible with mouse)

Dosant reviewed (commented):
Nice work! Was surprised to see `value` where possible as part of documentation! 

Some feedback I think we could try to improve: 

1. [ ] When I typed `{{` and then do auto-completion inside it results in `{{{{`. Would it be possible to avoid it? 
![Screenshot 2021-02-10 at 11 44 23](https://user-images.githubusercontent.com/7784120/107506468-6517de80-6b9e-11eb-9e1d-3e867e76cc11.png)
![Screenshot 2021-02-10 at 11 44 26](https://user-images.githubusercontent.com/7784120/107506471-66e1a200-6b9e-11eb-8c2e-31d15dabb6b9.png)


2. [ ] In older implementation longer URLs wrapped and this was convenient, is it possible to do something similar? I think currently It is more difficult to work with longer URLs: 
Before: 
![Screenshot 2021-02-10 at 12 30 23](https://user-images.githubusercontent.com/7784120/107506563-8678ca80-6b9e-11eb-9ae7-4b3fb5802c55.png)
After: 
![Screenshot 2021-02-10 at 12 50 39](https://user-images.githubusercontent.com/7784120/107506612-9a243100-6b9e-11eb-919f-7019a8753993.png)

3.  [ ] To my eye editor doesn't have enough contrast with the flyout background. Maybe it needs a border or darker background? cc @mdefazio 

4. [ ] URL highlight breaks for complex URLs, for example, like we have in Kibana: 
try this one:
```
{{kibanaUrl}}/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'{{event.from}}',to:'{{event.to}}'))&_a=(columns:!(_source),filters:{{rison context.panel.filters}},index:'{{context.panel.indexPatternId}}',interval:auto,query:(language:{{context.panel.query.language}},query:'{{context.panel.query.query}}'),sort:!())
```
![Screenshot 2021-02-10 at 12 54 30](https://user-images.githubusercontent.com/7784120/107506985-31898400-6b9f-11eb-8d5e-0a53975dc0f4.png)

if fixing this is a question of adjusting regex, then it would definitely worth improving 👍 

5. [ ] Looks like URL dropdown editor stuff was part of a main bundle :( (tech debt). With this pr this piece is growing significantly. Would it be possible to move some of that into async chunk? If it is not straightforward, then we should create an issue. 

6. [ ] One of the functional test is still failing? Didn't look into that. Just calling out, not sure if a legit regression.

mdefazio commented:
> To my eye editor doesn't have enough contrast with the flyout background. Maybe it needs a border or darker background?

This is a tough balance—if we darken the background it will likely not have enough contrast with the text inside it.  Perhaps a slightly darker border as you mentioned might help better define where the code editor is within the flyout.  Though I don't know that we do this elsewhere.  Maybe wait and see if others mention it as well?

streamich commented:
@mdefazio I've implemented both of your suggestions.

@ppisljar still looking into some of your comments:

- The autocompletion pops up when you press Cmd + Space or Monaco editor allows you to specify special characters on which the autocompletion pops up. Currently, those characters are specified as `['{', '/', '?', '&', '=']`. I can possibly tweak it to make it appear less often, will take a look.
- I have fixed the nested `{}` issue, so now if you have `{|}` or `{{|}}` it will not insert extra brackets.
- I'll see what I can do about autocompletion once you are inside the handlebars tokens, currently it has no special treatment for that.
- You can close the auto-completion popup with `Esc` button. The last meeting we discussed that showing auto-completion on mouse hover is not mandatory, but I'll see what I can do.

@Dosant  all your comments should be addressed:

- Fixed autocompletion with extra `{{{{`.
- Added line-wrapping.
- I haven't done anything for changing the background of the editor, as suggested by Michael let's see what others think. And another thing to note is that the Monaco editor component is shared, so such CSS changes will influence everyone.
- I've solved the complex URL autocompletion with a quick tweak by removing the green highlighting for the query parameter values, as it wasn't really useful anyway.
- I've made the drilldown config component load lazily.
- The functional tests were working, I think the recent change of adding a default URL (as there is no placeholder functionality in Monaco editor) breaks it, so in the functional test, I guess I just need to clean up the editor box, will fix it.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106362/)
* Commit: e4b6c4566b363e0b1c1acd970fee07cabad38cb5
* [Documentation Changes](https://kibana_88577.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/644bcbccd41aa7a226a48a3ac154a5a360c4d002) | [after](https://github.com/elastic/kibana/commit/e4b6c4566b363e0b1c1acd970fee07cabad38cb5) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 321 | 334 | +13 |
| `uiActionsEnhanced` | 120 | 124 | +4 |
| `urlDrilldown` | 29 | 33 | +4 |
| total |  |  | +21 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/644bcbccd41aa7a226a48a3ac154a5a360c4d002) | [after](https://github.com/elastic/kibana/commit/e4b6c4566b363e0b1c1acd970fee07cabad38cb5) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 342.8KB | 342.9KB | +150.0B |
| `uiActionsEnhanced` | 0.0B | 12.3KB | +12.3KB |
| total |  |  | +12.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/644bcbccd41aa7a226a48a3ac154a5a360c4d002) | [after](https://github.com/elastic/kibana/commit/e4b6c4566b363e0b1c1acd970fee07cabad38cb5) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 119.4KB | 127.3KB | +7.9KB |
| `uiActionsEnhanced` | 226.3KB | 217.5KB | -8.8KB |
| `urlDrilldown` | 42.5KB | 52.2KB | +9.7KB |
| total |  |  | +8.9KB |

<details><summary>Unknown metric groups</summary>

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/644bcbccd41aa7a226a48a3ac154a5a360c4d002) | [after](https://github.com/elastic/kibana/commit/e4b6c4566b363e0b1c1acd970fee07cabad38cb5) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 0 | 1 | +1 |


</details>

### History
* :broken_heart: [Build #105182](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/) failed 5b4fefebb980ee84157def88a3f6c6fa866057db
* :broken_heart: [Build #105042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/) failed c0de84fd76b93d0ad2ed5316bc540dd2759a344a
* :broken_heart: [Build #104783](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/) failed 0d3c826108c224a3b84da6b4a6d7ec4f311b231a
* :green_heart: [Build #104709](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/) succeeded 6f9189438c93e51122acb7050b4a2ddc0b73acc8
* :broken_heart: [Build #104648](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/) failed 1a525b0adb40528e6ae53135d1c12a605ed2c7f0

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104648/","number":"104648","commit":"1a525b0adb40528e6ae53135d1c12a605ed2c7f0"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/","number":"104709","commit":"6f9189438c93e51122acb7050b4a2ddc0b73acc8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/","number":"104783","commit":"0d3c826108c224a3b84da6b4a6d7ec4f311b231a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/","number":"105042","commit":"c0de84fd76b93d0ad2ed5316bc540dd2759a344a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/","number":"105182","commit":"5b4fefebb980ee84157def88a3f6c6fa866057db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106362/","number":"106362","commit":"e4b6c4566b363e0b1c1acd970fee07cabad38cb5"}]}
PIPELINE-->

streamich commented:
Zoomed with @ppisljar, we agreed that current PR is good to be merged and in a separate PR I will work to improve:

- Autocompletion should not open once user types regular characters after a special character, for example, after typing `c` after `.`, in `www.example.c|`.
- Autocompletion should not open, or open more smartly, once user's cursor is in the block of Handlebars variable.

ppisljar reviewed (approved):
LGTM

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106511/)
* Commit: 09cbfe16fef32c386b890207457cfaa278ce5429
* [Documentation Changes](https://kibana_88577.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/c1d1b2b453465a2d788110b7847e15c31be95b28) | [after](https://github.com/elastic/kibana/commit/09cbfe16fef32c386b890207457cfaa278ce5429) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 321 | 334 | +13 |
| `uiActionsEnhanced` | 120 | 124 | +4 |
| `urlDrilldown` | 29 | 33 | +4 |
| total |  |  | +21 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/c1d1b2b453465a2d788110b7847e15c31be95b28) | [after](https://github.com/elastic/kibana/commit/09cbfe16fef32c386b890207457cfaa278ce5429) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 342.8KB | 342.9KB | +150.0B |
| `uiActionsEnhanced` | 0.0B | 12.3KB | +12.3KB |
| total |  |  | +12.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/c1d1b2b453465a2d788110b7847e15c31be95b28) | [after](https://github.com/elastic/kibana/commit/09cbfe16fef32c386b890207457cfaa278ce5429) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 119.4KB | 127.3KB | +7.9KB |
| `uiActionsEnhanced` | 226.3KB | 217.5KB | -8.8KB |
| `urlDrilldown` | 42.5KB | 52.2KB | +9.7KB |
| total |  |  | +8.9KB |

<details><summary>Unknown metric groups</summary>

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/c1d1b2b453465a2d788110b7847e15c31be95b28) | [after](https://github.com/elastic/kibana/commit/09cbfe16fef32c386b890207457cfaa278ce5429) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 0 | 1 | +1 |


</details>

### History
* :green_heart: [Build #106362](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106362/) succeeded e4b6c4566b363e0b1c1acd970fee07cabad38cb5
* :broken_heart: [Build #105182](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/) failed 5b4fefebb980ee84157def88a3f6c6fa866057db
* :broken_heart: [Build #105042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/) failed c0de84fd76b93d0ad2ed5316bc540dd2759a344a
* :broken_heart: [Build #104783](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/) failed 0d3c826108c224a3b84da6b4a6d7ec4f311b231a
* :green_heart: [Build #104709](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/) succeeded 6f9189438c93e51122acb7050b4a2ddc0b73acc8

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104709/","number":"104709","commit":"6f9189438c93e51122acb7050b4a2ddc0b73acc8"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/","number":"104783","commit":"0d3c826108c224a3b84da6b4a6d7ec4f311b231a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/","number":"105042","commit":"c0de84fd76b93d0ad2ed5316bc540dd2759a344a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/","number":"105182","commit":"5b4fefebb980ee84157def88a3f6c6fa866057db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106362/","number":"106362","commit":"e4b6c4566b363e0b1c1acd970fee07cabad38cb5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106511/","number":"106511","commit":"09cbfe16fef32c386b890207457cfaa278ce5429"}]}
PIPELINE-->

Dosant commented on x-pack/plugins/drilldowns/url_drilldown/public/lib/url_drilldown.tsx:
```diff
@@ -90,21 +96,30 @@ export class UrlDrilldown implements Drilldown<Config, ActionContext, ActionFact
   public readonly createConfig = () => ({
-    url: { template: '' },
+    url: {
+      template: 'Example: https://example.com/?{{event.key}}={{event.value}}',
```
I think that because of how we don't have a real placeholder, but a "default" URL instead it is not a good idea to have `Example ` prefix 🤔 

Because now our _default_ URL template is invalid

cc @mdefazio

streamich replied:
I agree with @Dosant, it feels more natural to have a real URL as that is not a placeholder but an example. I've reverted back to have it without "Example: " prefix, unless you have a strong preference @mdefazio.

Dosant reviewed (approved):
Thanks for addressing earlier feedback 🚀

mdefazio reviewed (approved):
Thanks for making the edits!

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106624/)
* Commit: fcaa6daa029422e7ad32f28c8591575b3429b7f7
* [Documentation Changes](https://kibana_88577.docs-preview.app.elstc.co/diff)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/42e11e6763472a1e25c0854506a618fe06c33bde) | [after](https://github.com/elastic/kibana/commit/fcaa6daa029422e7ad32f28c8591575b3429b7f7) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 321 | 334 | +13 |
| `uiActionsEnhanced` | 120 | 124 | +4 |
| `urlDrilldown` | 29 | 33 | +4 |
| total |  |  | +21 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/42e11e6763472a1e25c0854506a618fe06c33bde) | [after](https://github.com/elastic/kibana/commit/fcaa6daa029422e7ad32f28c8591575b3429b7f7) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 342.8KB | 342.9KB | +150.0B |
| `uiActionsEnhanced` | 0.0B | 12.3KB | +12.3KB |
| total |  |  | +12.4KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/42e11e6763472a1e25c0854506a618fe06c33bde) | [after](https://github.com/elastic/kibana/commit/fcaa6daa029422e7ad32f28c8591575b3429b7f7) | diff |
| --- | --- | --- | --- |
| `kibanaReact` | 119.4KB | 127.3KB | +8.0KB |
| `uiActionsEnhanced` | 226.3KB | 217.5KB | -8.8KB |
| `urlDrilldown` | 42.5KB | 52.2KB | +9.7KB |
| total |  |  | +8.9KB |

<details><summary>Unknown metric groups</summary>

#### async chunk count
| id | [before](https://github.com/elastic/kibana/commit/42e11e6763472a1e25c0854506a618fe06c33bde) | [after](https://github.com/elastic/kibana/commit/fcaa6daa029422e7ad32f28c8591575b3429b7f7) | diff |
| --- | --- | --- | --- |
| `uiActionsEnhanced` | 0 | 1 | +1 |


</details>

### History
* :green_heart: [Build #106511](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106511/) succeeded 09cbfe16fef32c386b890207457cfaa278ce5429
* :green_heart: [Build #106362](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106362/) succeeded e4b6c4566b363e0b1c1acd970fee07cabad38cb5
* :broken_heart: [Build #105182](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/) failed 5b4fefebb980ee84157def88a3f6c6fa866057db
* :broken_heart: [Build #105042](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/) failed c0de84fd76b93d0ad2ed5316bc540dd2759a344a
* :broken_heart: [Build #104783](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/) failed 0d3c826108c224a3b84da6b4a6d7ec4f311b231a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/104783/","number":"104783","commit":"0d3c826108c224a3b84da6b4a6d7ec4f311b231a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105042/","number":"105042","commit":"c0de84fd76b93d0ad2ed5316bc540dd2759a344a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/105182/","number":"105182","commit":"5b4fefebb980ee84157def88a3f6c6fa866057db"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106362/","number":"106362","commit":"e4b6c4566b363e0b1c1acd970fee07cabad38cb5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106511/","number":"106511","commit":"09cbfe16fef32c386b890207457cfaa278ce5429"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/106624/","number":"106624","commit":"fcaa6daa029422e7ad32f28c8591575b3429b7f7"}]}
PIPELINE-->
